### PR TITLE
Shopping cart us23

### LINKED ISSUE #34: User Story 23, Adding Item Quantity to Cart
As a visitor
When I have items in my cart
And I visit my cart
Next to each item in my cart
I see a button or link to increment the count of items I want to purchase
I cannot increment the count beyond the item's inventory size

### PR DESCRIPTION
**Pull request by priyapower**
closes #34 

 **Contributors**
- Priya

 **What was changed:**
- Creates inventory checks when we do an add to cart
- Created a way to plus one to an item in your cart
- Creates inventory checks when we update an item in the cart
- Update pre-written test to pass with new inventory checks

 **Any tech notes (if applicable)**
- The cart view pages and checkout (order/new) view pages are aligning funky

### CODE REVIEW

**Comment by priyapower** [comment]
I realized I didn't finish this user story!! It needs an update for a logged in regular, merchant user and restrictions for admin user

**Comment by priyapower** [comment]
> I realized I didn't finish this user story!! It needs an update for a logged in regular, merchant user and restrictions for admin user

This is updated now!!! It can be reviewed! Sorry I didn't catch my empty tests before

File: app/controllers/cart_controller.rb [hunk 1]
```diff
@@ -1,15 +1,36 @@
...
     item = Item.find(params[:item_id])
     cart.add_item(item.id.to_s)
-    flash[:success] = "#{item.name} was successfully added to your cart"
-    redirect_to "/items"
+    if cart.contents[item.id] == 0
+      cart.empty_cart
+      flash[:failure] = "There are limited supplies in inventory - this item has reached it's max! Please try again later."
+    else
+      flash[:success] = "#{item.name} was successfully added to your cart"
+      redirect_to "/items"
```
Comment (LHJE) [inline app/controllers/cart_controller.rb:12]: Nice logic!

File: app/views/cart/show.html.erb [hunk 2]
```diff
@@ -15,8 +16,14 @@
       <td>
         <p align = "center">
           <span>
-            <img align = "center" id="cart-image" src= <%= item.image %>>
-            <%=link_to item.name, "/items/#{item.id}" %>
+            <ul><img align = "center" id="cart-image" src= <%= item.image %> style= "max-width:200px;width:100%" ></ul>
```
Comment (LHJE) [inline app/views/cart/show.html.erb:19]: Nice drying up of the code!

File: spec/features/cart/add_spec.rb [hunk 3]
```diff
@@ -2,34 +2,172 @@
...
+      end
 
-      within 'nav' do
-        expect(page).to have_content("Cart: 2")
+      it "can NOT see the cart as an admin" do
+        visit "/items/#{@quantity_test_object_admin.id}"
+        expect(page).to_not have_button("Add To Cart")
       end
     end
   end
```
Comment (LHJE) [inline spec/features/cart/add_spec.rb:173]: All of these edits look great!

**Review by LHJE — APPROVED** [review]
Looks Great!

File: app/views/cart/show.html.erb (hunk 2 above)
Comment (priyapower) [reply app/views/cart/show.html.erb:19]: I'm still struggling with html/css, so I am working on making updates as I see them after as ```save_and_open_page```

File: spec/features/cart/add_spec.rb (hunk 3 above)
Comment (priyapower) [reply spec/features/cart/add_spec.rb:173]: Woo! Thanks!

**Pull request merged**